contracts/RewardsLiquidator.vy

```diff
@@ -147,98 +239,156 @@ def _get_eth_anchor_price() -> uint256:
+            method_id("exactInputSingle((address,address,uint24,address,uint256,uint256,uint256,uint160))"),
             convert(WETH_TOKEN, bytes32),
-            convert(UST_TOKEN, bytes32)
+            convert(USDC_TOKEN, bytes32),
+            convert(self.uniswap_usdc_pool_fee, bytes32), #pool fee
+            convert(usdc_recipient, bytes32), #recipient
+            convert(block.timestamp, bytes32), #deadline
+            convert(eth_amount_in, bytes32),
+            convert(usdc_amount_out_min, bytes32),
+            convert(0, bytes32), #sqrtPriceLimitX96
         ),
         value=eth_amount_in,
-        max_outsize=128
+        max_outsize=32
     )
-    # The return type is uint256[] and the actual length of the array is 2, so the
-    # layout of the data is (pad32(64) . pad32(2) . pad32(arr[0]) . pad32(arr[1])).
-    # We need the second array item, thus the byte offset is 96.
-    return convert(extract32(result, 96), uint256)
+    return convert(result, uint256)
 
 
 @internal
 @pure
-def _get_min_amount_out(amount_in: uint256, price: uint256, max_diff_percent: uint256) -> uint256:
-    amount_out: uint256 = (amount_in * price) / 10**18
+def _get_min_amount_out(
+    amount: uint256,
+    price: uint256,
+    max_diff_percent: uint256,
+    decimal_token_in: uint256,
+    decimal_token_out: uint256
+) -> uint256:
+    amount_out: uint256 = (amount * price) / (10 ** decimal_token_in) # = (amount * (10 ** (18 - decimal_token_in)) * price) / 10 ** 18
     min_mult: uint256 = 10**18 - max_diff_percent
-    return (amount_out * min_mult) / 10**18
-
+    return (amount_out * min_mult) / (10 ** (36 - decimal_token_out)) # = ((amount_out * min_mult) / 10**18) / (10 ** (18 - decimal_token_out))
 
+# 1) stETH -> ETH (Curve)
+# 2) ETH -> USDC (Uniswap v3)
+# 3) USDC -> UST (Curve)
 @external
 def liquidate(ust_recipient: address) -> uint256:
     assert msg.sender == self.vault, "unauthorized"
 
     steth_amount: uint256 = ERC20(STETH_TOKEN).balanceOf(self)
     assert steth_amount > 0, "zero stETH balance"
 
-    steth_anchor_price: uint256 = self._get_steth_anchor_price()
-
+    # steth -> eth
+    steth_eth_price: uint256 = self._get_chainlink_price(CHAINLINK_STETH_ETH_FEED, False)
     min_eth_amount: uint256 = self._get_min_amount_out(
         steth_amount,
-        steth_anchor_price,
-        self.max_steth_price_difference_percent
+        steth_eth_price,
+        self.max_steth_eth_price_difference_percent,
+        STETH_TOKEN_DECIMALS,
+        WETH_TOKEN_DECIMALS
     )
 
-    ERC20(STETH_TOKEN).approve(CURVE_STETH_POOL, steth_amount)
+    ERC20(STETH_TOKEN).approve(self.curve_steth_pool, steth_amount)
 
-    eth_amount: uint256 = CurveStableSwap(CURVE_STETH_POOL).exchange(
-        CURVE_STETH_INDEX,
-        CURVE_ETH_INDEX,
+    eth_amount: uint256 = CurvePool(self.curve_steth_pool).exchange(
+        self.curve_steth_index,
+        self.curve_eth_index,
         steth_amount,
         min_eth_amount
```

Comment:
I've one idea: what if we set min return amount to 0 everywhere? This way, checks would still be performed by our code, a transaction will anyway fail if slippage is too high, but it will fail with some readable revert code.

Reply:
should work, this is even more correct, since the calculated `min_amount` is based only on the Chainlink price, which in general does not take into account the state of the pool at the moment. So we don't care about pool internal price

Reply:
I believe that this can even reduce slippage a bit

Reply:
It shouldn't reduce slippage since all that passing min amount does is that the pool checks whether the resulting amount is at least the passed one and reverts the transaction if not, e.g. see https://github.com/curvefi/curve-contract/blob/master/contracts/pools/steth/StableSwapSTETH.vy#L453. But sometimes (e.g. in the Sushi's case) this check is performed without any revert message.